corpus/segments.py

```diff
@@ -49,7 +50,16 @@ def run(self):
         for idx, segments in enumerate(chunks(all_segments, self.num_segments)):
             with open(self.out_single_segment_files[idx + 1].get_path(), "wt") as segment_file:
                 for segment in segments:
-                    segment_file.write(segment.fullname() + "\n")
+                    name = segment.fullname() + "\n"
+                    name = name[name.startswith(self.remove_prefix) and len(self.remove_prefix) :]
+                    segment_file.write(name)
+
+    @classmethod
+    def hash(cls, kwargs):
+        d = dict(**kwargs)
+        if not d["remove_prefix"]:
+            d.pop("remove_prefix")
+        return super().hash(d)
```

Comment:
I think it would be cleaner to use `__sis_hash_exclude__ = {"remove_prefix": ""}` at the beginning of the job.

Reply:
Yes, if new parameters are introduced this is the way to go.